### DISCUSSION TITLE
How to detect that computation is finished?

### DISCUSSION THREAD

**Discussion by danilo-89** [mobxjs/mobx#3489] [Q&A]
I have a computed that does some complex, time intense, computations.

Because of the first rule of computed values ([source](https://mobx.js.org/computeds.html)):
> computed should not have side effects or update other observables

what is the best way to detect when computed is finished with computing?

**example:**
store:
  ```
        value1 = 2
        value2 = 1

	constructor() {
		makeObservable(this, {
			value1: observable,
			value1: observable,
			computedValue: computed,
		})
	}

	computedValue = () => {
		const result = // some complex, time intense, computations with value1 and value2 that repeats often
		return result
	};
```

html/js
```
	// detect if computing is still in progress and display something else
	<div>loading....</div>
	// when computing finished display
	<div>{computedValue}</div>
```

**Comment by kubk** [comment]
I am lacking the context here to help. Could you describe the exact issue to avoid the XY problem? https://xyproblem.info/

**Reply by danilo-89** [reply]
question updated, hope it's clearer now, thank you

**Reply by mweststrate** [reply]
computeds are executed synchronously, so there is no way to 'observe' that the computeds are executig or to propagate that to the UI. If the computation is so expensive you'd want to 'time slice' it, it's probably safer to manage it in a reaction, split it in small chunks, and write the output somewhere meaningfull, otherwise the JS engine will just be blocked and can't do anything (like scrolling, resizing or communicating 'loading...' states.
